Fix use of allow_tensor_metadata in view variable creation

albanD opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57058 Remove unused forward AD flag
* #57057 Improve perf for forward AD view handling
* #57152 fix InplaceOrView implementation for manual functions
* **#57069 Fix use of allow_tensor_metadata in view variable creation**

Differential Revision: [D28071507](https://our.internmc.facebook.com/intern/diff/D28071507)

albanD committed:
Fix use of allow_tensor_metadata in view variable creation

[ghstack-poisoned]

albanD committed:
Update on "Fix use of allow_tensor_metadata in view variable creation"




[ghstack-poisoned]

ailzhang reviewed (approved):
Nice catch! I'm actually curious what's the uses cases for `allow_tensor_metadata_change`, do we have a pointer to a note? :D

albanD committed:
Update on "Fix use of allow_tensor_metadata in view variable creation"




[ghstack-poisoned]

albanD committed:
Update on "Fix use of allow_tensor_metadata in view variable creation"


Differential Revision: [D28071507](https://our.internmc.facebook.com/intern/diff/D28071507)

[ghstack-poisoned]
